lib/galaxy/tools/parameters/basic.py

```diff
@@ -1114,11 +1116,16 @@ def get_column_list(self, trans, other_values):
             if isinstance(dataset, trans.app.model.HistoryDatasetCollectionAssociation):
                 dataset = dataset.to_hda_representative()
             # Columns can only be identified if metadata is available
-            if not hasattr(dataset, 'metadata') or not hasattr(dataset.metadata, 'columns') or not dataset.metadata.columns:
+            if not hasattr(dataset, 'metadata') or not hasattr(dataset.metadata, 'columns'):
                 return []
             # Build up possible columns for this dataset
             this_column_list = []
-            if self.numerical:
+            # Valid column-based datasets contain at least 1 column if that column has not been
+            # specified we prepopulate the selector assuming that the datasets is not ready yet.
+            if dataset.metadata.columns is None:
+                for i in range(0, MAX_DEFAULT_COLUMNS):
+                    this_column_list.append(str(i + 1))
```

Comment:
You can replace the last 2 lines with
```python
                this_column_list = list(map(str, range(1, MAX_DEFAULT_COLUMNS + 1)))
```
which takes around half the time.
Similarly for lines 1134-1135.

Reply:
Thx!